src/synced_services/toggl/toggl_synced_service.ts

```diff
@@ -280,103 +274,58 @@ export class TogglTrackSyncedService implements SyncedService {
-            timeEntry['id'],
-            timeEntry['pid'],
-            timeEntry['description'],
-            new Date(timeEntry['start']),
-            new Date(timeEntry['end']),
-            timeEntry['dur'],
-            timeEntry['tags'],
-            new Date(timeEntry['updated']),
-          ),
-        );
-      });
-
-      totalCount = response.body?.total_count;
-    } while (queryParams.page * this._responseLimit < totalCount);
+    const end = new Date();
+    end.setFullYear(9999);
 
-    return entries;
-  }
-
-  async getTimeEntryById(id: number | string, start?: Date): Promise<TimeEntry | null> {
     const queryParams = {
-      time_entry_ids: id,
-      since: start?.toISOString(),
-      workspace_id: this._serviceDefinition.config.workspace?.id,
-      user_ids: this._serviceDefinition.config.userId,
-      user_agent: 'Timer2Ticket',
-      page: 0,
+      start_date: start?.toISOString(),
+      end_date: end?.toISOString(),
     };
 
     const entries: TogglTimeEntry[] = [];
 
-    // time entries via reports (requesting only one object => do not paginate)
-
     let response;
 
     try {
       response = await this._retryAndWaitInCaseOfTooManyRequests(
           superagent
-              .get(this._reportsUri)
+              .get(this._timeEntriesUri)
               .query(queryParams)
               .auth(this._serviceDefinition.apiKey, 'api_token')
       );
     } catch (ex: any) {
-      this.handleResponseException(ex, 'getTimeEntryById');
-      return null;
+      this.handleResponseException(ex, 'getTimeEntries')
+      return [];
     }
 
 
-    response.body?.data.forEach((timeEntry: never) => {
-      entries.push(
-        new TogglTimeEntry(
-          timeEntry['id'],
-          timeEntry['pid'],
-          timeEntry['description'],
-          new Date(timeEntry['start']),
-          new Date(timeEntry['end']),
-          timeEntry['dur'],
-          timeEntry['tags'],
-          new Date(timeEntry['updated']),
-        ),
-      );
+    response.body?.forEach((timeEntry: never) => {
+      if(timeEntry['workspace_id'] === this._serviceDefinition.config.workspace?.id.toString()) {
+        entries.push(
+            new TogglTimeEntry(
+                timeEntry['id'],
+                timeEntry['project_id'],
+                timeEntry['description'],
+                new Date(timeEntry['start']),
+                new Date(timeEntry['stop']),
+                timeEntry['duration'] * 1000,
+                timeEntry['tags'],
+                new Date(timeEntry['at']),
+            ),
+        );
+      }
     });
 
-    return entries.length === 1 ? entries[0] : null;
+    return entries;
+  }
+
+  async getTimeEntryById(id: number | string, start?: Date): Promise<TimeEntry | null> {
+
+    const entries = this.getTimeEntries(start);
+
+    const entry = (await entries).filter((entry: TimeEntry) => entry.id.toString() === id.toString());
+
+    return entry.length === 1 ? entry[0] : null;
```

Comment:
Tento krok je predpokladam z dovodu DISABLe odpovedajuceho EP. Neviem, ci je to castou tvojej prace, ale tu by sa mi hodila nejaka analyza toho, ze tato zmena uplne nezahlti computing power (prechadzame z callu na 1 EP, kde dostaneme max 1 vysledok => call so 100kami vysledkov, ktory nasledne filtrujeme na urovni programu).

Reply:
Pride mi vyhodnejsie pouzit https://developers.track.toggl.com/docs/api/time_entries/index.html#get-timeentries, ktore taktiez podporuje start_date a end_date filtre, navyse priamo v repsonse dostavame Tags, takze nemusime prevolavat getAllTags EP.
Vidis nejaku vyhodu v pouziti /reports namiesto /timeEntries?

Reply:
Hlavní výhoda je v tom, že /reports umožňuje filtrovat podle id časového záznamu. Ten endpoint, který zmiňuješ, tam byl fakticky implementován před mou úpravou

Reply:
Okj, nevsimol som si filter na time_entry_id, v tomto pripade mi to dava zmysel, vdaka.